packages/next/src/views/List/renderListViewSlots.tsx

```diff
@@ -75,13 +79,31 @@ export const renderListViewSlots = ({
     })
   }
 
-  if (collectionConfig.admin.components?.beforeListTable) {
-    result.BeforeListTable = RenderServerComponent({
-      clientProps: clientProps satisfies BeforeListTableClientProps,
-      Component: collectionConfig.admin.components.beforeListTable,
-      importMap: payload.importMap,
-      serverProps: serverProps satisfies BeforeListTableServerPropsOnly,
-    })
+  // Handle beforeListTable with optional banner
+  const existingBeforeListTable = collectionConfig.admin.components?.beforeListTable
+    ? RenderServerComponent({
+        clientProps: clientProps satisfies BeforeListTableClientProps,
+        Component: collectionConfig.admin.components.beforeListTable,
+        importMap: payload.importMap,
+        serverProps: serverProps satisfies BeforeListTableServerPropsOnly,
+      })
+    : null
+
+  // Create banner for document not found
+  const notFoundBanner = notFoundDocId ? (
+    <Banner type="error">
+      {serverProps.i18n.t('error:loadingDocument', { id: notFoundDocId })}
```

Comment:
I used `loadingDocument` because it already existed.
It says: `There was a problem loading the document with ID of {{id}}`.

Maybe something like this could be better?:
`documentNotFound: 'The document with ID {{id}} does not exist or may have been deleted.',`